pkg/controller/operators/olm/operator.go

```diff
@@ -550,14 +567,66 @@ func (a *Operator) syncObject(obj interface{}) (syncError error) {
 		"self":      metaObj.GetSelfLink(),
 	})
 
-	// Requeue all owner CSVs
-	if ownerutil.IsOwnedByKind(metaObj, v1alpha1.ClusterServiceVersionKind) {
-		logger.Debug("requeueing owner csvs")
-		a.requeueOwnerCSVs(metaObj)
+	switch metaObj.(type) {
+	case *corev1.ServiceAccount:
+		if err := a.opClient.DeleteServiceAccount(metaObj.GetNamespace(), metaObj.GetName(), &metav1.DeleteOptions{}); err != nil {
+			logger.WithError(err).Warn("cannot delete service account")
+			syncError = err
+		}
+		logger.Debugf("Deleted apiservice %v due to no owning CSV", metaObj.GetName())
+	case *rbacv1.ClusterRole:
+		if err := a.opClient.DeleteClusterRole(metaObj.GetName(), &metav1.DeleteOptions{}); err != nil {
+			logger.WithError(err).Warn("cannot delete cluster role")
+			syncError = err
+		}
+		logger.Debugf("Deleted cluster role %v due to no owning CSV", metaObj.GetName())
+	case *rbacv1.ClusterRoleBinding:
+		if err := a.opClient.DeleteClusterRoleBinding(metaObj.GetName(), &metav1.DeleteOptions{}); err != nil {
+			logger.WithError(err).Warn("cannot delete cluster role binding")
+			syncError = err
+		}
+		logger.Debugf("Deleted cluster role binding %v due to no owning CSV", metaObj.GetName())
+	}
+
+	return
+}
+
+func (a *Operator) syncObject(obj interface{}) (syncError error) {
+	// Assert as metav1.Object
+	metaObj, ok := obj.(metav1.Object)
+	if !ok {
+		syncError = errors.New("object sync: casting to metav1.Object failed")
+		a.logger.Warn(syncError.Error())
+		return
 	}
+	logger := a.logger.WithFields(logrus.Fields{
+		"name":      metaObj.GetName(),
+		"namespace": metaObj.GetNamespace(),
+		"self":      metaObj.GetSelfLink(),
+	})
 
 	// Requeues objects that can't have ownerrefs (cluster -> namespace, cross-namespace)
 	if ownerutil.IsOwnedByKindLabel(metaObj, v1alpha1.ClusterServiceVersionKind) {
+		name, ns, ok := ownerutil.GetOwnerByKindLabel(metaObj, v1alpha1.ClusterServiceVersionKind)
+		if !ok {
+			logger.Error("unexpected owner label retrieval failure")
+		}
+		_, err := a.lister.OperatorsV1alpha1().ClusterServiceVersionLister().ClusterServiceVersions(ns).Get(name)
+		if !k8serrors.IsNotFound(err) {
+			logger.Debug("requeueing owner csvs from owner label")
+			a.requeueOwnerCSVs(metaObj)
+		} else {
+			switch metaObj.(type) {
+			case *corev1.ServiceAccount, *rbacv1.ClusterRole, *rbacv1.ClusterRoleBinding:
+				a.objGCQueueSet.Requeue(ns, metaObj.GetName())
```

Comment:
If the goal is to send a `ResourceEvent`, this will not work as intended since it constructs a key manually. It may be helpful to add a `RequeueEvent` method to the `ResourceQueue` type.